### DISCUSSION TITLE
Adding QOI support

### DISCUSSION THREAD

**Discussion by LuisAlfredo92** [SixLabors/ImageSharp#2403] [Ideas] [answered]
QOI is a image lossless compression algorithm in O(n) time. It's very simple and easy to implement and in my opinion it should be more adopted by developers.

- All the information can be found here: https://qoiformat.org/
- It's open source: https://github.com/phoboslab/qoi
- It has C# implementations with encoder and decoder: https://github.com/NUlliiON/QoiSharp

**Comment by JimBobSquarePants** [comment]
I don't see why not. It's a simple enough format and has garnered broad support. 👍

Would you be interested in doing the work yourself?

**Reply by LuisAlfredo92** [reply, marked as answer]
Sure, it'd be nice, but I'll have to study the project structure to know what files and where I have to add them

EDIT: Working on that #2446

**Reply by JimBobSquarePants** [reply]
That's awesome. Thanks!

The formats all exist here. 

https://github.com/SixLabors/ImageSharp/tree/5283d7736f3546c67b63d8a95d26830bb83f4836/src/ImageSharp/Formats

The best approach is to inherit `ImageDecoder` and `ImageEncoder` as that saves a bunch of work. You're inherited types should be stateless with all the actual work delegated to stateful "Core" implementations. 

Any questions, jusy open a draft PR and we can help guide you.
